src/main/java/com/github/davidmoten/rx/jdbc/RegexMappableResultSet.java

```diff
@@ -0,0 +1,830 @@
+package com.github.davidmoten.rx.jdbc;
+
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.*;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class RegexMappableResultSet implements ResultSet {
+
+    private final ResultSet rs;
+    private final int columnCount;
+    private final String[] columnLabels;
+    private final HashMap<String,Integer> cachedIndices = new HashMap<>(); //-1 means unfound
+
+    private RegexMappableResultSet(ResultSet rs) {
+        try {
+            this.rs = rs;
+
+            columnCount = rs.getMetaData().getColumnCount();
+            columnLabels = new String[columnCount];
+
+            for (int i=1;i<=columnCount;i++) {
+                columnLabels[i-1] = rs.getMetaData().getColumnName(i);
+            }
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public int findColumn(String columnRegex) throws SQLException {
+            Integer foundIndex = cachedIndices.get(columnRegex);
+            if (foundIndex == null) {
+                    foundIndex = findColumnIndex(columnRegex);
+                    cachedIndices.put(columnRegex, foundIndex);
+                }
+            return foundIndex;
+    }
+    private Integer findColumnIndex(String columnRegex) {
+            for (int i=0;i<columnCount;i++) {
+                    if (columnLabels[i].matches(columnRegex)) {
+                            return Integer.valueOf(i);
+                        }
+                }
+            return Integer.valueOf(-1);
```

Comment:
Okay I'll implement that later today. If I do two implementations I may have to revise the exception strategies for unfound columns on `RegexMappableResultSet` otherwise it would logically clash with this `NullableIfUnfoundResultSet`. My reasoning so far indicates the `RegexMappableResultSet` must have some notion of nullability or else the getters() will only throw exceptions if columnLabels are unfound.